### DISCUSSION TITLE
Progress bar con texto similar al block de Community en Mistica?

### DISCUSSION THREAD

**Discussion by MontseTelefonica** [Telefonica/mistica-design#1692] [Proposals]
### Goal

Necesitamos disponer de un componente de progress bar con texto que usar para varios casos, pero el más urgente es el test A/B de UK de pelochos contra cards. También se avecina una necesidad similar para la new app de Moves y su pantalla de consumos, que está llena de progress bars y para las nuevas versiones de O2 tambien se usan muchas cards con progress bar.

<img width="351" alt="image" src="https://github.com/Telefonica/mistica-design/assets/95479574/302e45e6-7f89-4310-a263-da7ce7c9a32d">

![Captura de pantalla 2024-03-06 a las 17 00 02](https://github.com/Telefonica/mistica-design/assets/95479574/8edb8152-8c74-4385-8b60-b1ce38ac3202)

### Context

Creo que por su alta frecuencia de uso ahora mismo sería necesario tener un componente propio y además no se nos permite usar el de community en desarrollo, aunque en diseño se esté usando.

### Impact

Crear este componente de forma global con varias opciones de configuración ayudaría a Mistificar elementos bastardos a día de hoy y poder dar soporte a los nuevos diseños.

### Discussion Questions

Es posible que esta necesidad sea absorbida por Mistica en un corto periodo de tiempo?

**Comment by aweell** [comment]
Buenas @MontseTelefonica!

> Crear este componente de forma global con varias opciones de configuración

Por entender un poco todos los escenarios en los que se está utilizando ese componente, que se ha identificado como "opciones de configuración" necesarias?

A priori solo con estos dos ejemplos intuyo que:

* Tamaño de textos
* Color de textos
* Alineamiento tanto vertical como horizontal de lo que podríamos llamar helper texts de currentValue y maxValue
* Que exista o no una "label" en la progress (o mas de una según veo en periodo de consumos)

**Comment by MontseTelefonica** [comment]
Correcto, esos son los elementos identificados hasta ahora. Necesitamos flexibilidad en ese componente para que pueda vivir solo en un layout o dentro de una card. Y dado que el mayor challenge puede venir de Moves, sería ideal poder replicar al detalle ambas opciones (Moves y O2UK) con el mismo componente.

**Reply by aweell** [reply]
En el caso del label de la progress por ejemplo:


![Screenshot 2024-03-07 at 16 56 32](https://github.com/Telefonica/mistica-design/assets/44420072/a9893bbe-1b6f-4ded-a513-12f0a9dda295)
![Screenshot 2024-03-07 at 16 56 36](https://github.com/Telefonica/mistica-design/assets/44420072/a4a6b49c-9ce3-4d77-a6ed-4c8a42d91eb7)
![Screenshot 2024-03-07 at 16 58 23](https://github.com/Telefonica/mistica-design/assets/44420072/63b1d1c1-61f9-44d0-9031-7b2448819ded)


En el primero por ejemplo es el title2 el que actúa como label de la progress, sin embargo en el segundo tiene su "propia label" (pensando que idealmente esa podría ser la que trae el componente por defecto), en el tercer caso es el title de la card. Parece ser algo muy variable y dependiente del contexto en el que se incluya.

Si lo comparamos por ejemplo con los inputs que es algo que siempre se comporta de la misma manera.

**Comment by MontseTelefonica** [comment]
Adjunto más necesidades localizadas según la propuesta de Moves:
<img width="323" alt="image" src="https://github.com/Telefonica/mistica-design/assets/95479574/8d534312-60ef-4f1b-8825-a56bf96d043d">
-Estilo de barra "ilimitado" 
-permitir 2 barras en paralelo (esto creo que no deberíamos permitirlo) pero lo dejo como necesidad detectada

**Reply by aweell** [reply]
El caso de las dos barras en paralelo definitivamente es algo que está fuera de la definición del componente, igual que los puedo stackear verticalmente en un slot podría hacerlo horizontalmente (aunque coincido que puede no ser lo ideal en determinados tamaños de pantalla)

**Comment by MontseTelefonica** [comment]
Quizás podamos intentar convencer a Moves de ir por algo así (opción de la derecha):
<img width="676" alt="image" src="https://github.com/Telefonica/mistica-design/assets/95479574/75f9f3ab-9d85-4195-bc3d-72e0b73688ca">

**Comment by MontseTelefonica** [comment]
Otra duda que tengo es que estructura es mejor: 
A o B
<img width="593" alt="image" src="https://github.com/Telefonica/mistica-design/assets/95479574/e1f760d4-f0cf-4e9b-8b49-bd537f1e0f5a">

**Comment by HectorSancho** [comment]
¡Hola! yo en este caso veo que es más parecido al antiguo slot que teníamos de progress pero con los valores invertidos.
Al cambiar tanto el tamaño del text-preset o el posicionamiento del otro elemento de text secondary, me cuesta verlo core. 

Dejo un listado de los motivos que me vienen a la cabeza:

- Que ocurre si un producto quiere invertir el posicionamiento
- Cambiar el tamaño, color o estilo del dato
- El uso de la progress en escenarios que no tengan nada que ver, esta estructura no es de utilidad (funnel o indicador de progreso de pasos o el track de un envío por ejemplo)

Sé lo que aporta el hecho que sea core para el CMS pero creo que no es buena solución a nivel de DS. La vía es tener los slots en algún sitio en el que su uso sea legítimo, pero sin meterlas en un componente core.

Los slots son Mística de forma legítima.

**Reply by MontseTelefonica** [reply]
Lo ves como candidato a componente extended? @HectorSancho ?

**Reply by HectorSancho** [reply]
Candidato es, si no vemos la forma de darle una definición core más general. De todas formas lo vemos y discutimos al detalle en core :)

**Reply by wyunreal** [reply]
Hola, al final que conclusiones tenéis en core ?

**Reply by lauraocana** [reply]
Continuamos con Novum Extended tal y como se acordó en diciembre. Thanks!!

**Comment by yceballost** [comment]
**Componente extended**

No tiene sentido incluir en Mística componentes con lógica de producto
